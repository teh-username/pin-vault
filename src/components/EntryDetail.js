import React from 'react';
import { StyleSheet, View, Alert } from 'react-native';
import { Button, Text } from 'react-native-elements';

export const ButtonRow = ({ onEditPress, onDeletePress }) => (
  <View style={styles.buttonRow}>
    <Button
      title="Edit Code"
      icon={{ name: 'mode-edit' }}
      onPress={onEditPress}
      buttonStyle={styles.buttonStyle}
    />
    <Button
      title="Delete Entry"
      icon={{ name: 'delete' }}
      onPress={onDeletePress}
      buttonStyle={styles.buttonStyle}
    />
  </View>
);

const EntryDetail = ({ name, code, onEditPress, onDeletePress }) => (
  <View style={styles.container}>
    <Text h1>{name}</Text>
    <Text h1>{code}</Text>
    <ButtonRow onEditPress={onEditPress} onDeletePress={onDeletePress} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    top: 30,
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  buttonRow: {
    flex: 1,
    top: 30,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  buttonStyle: {
    backgroundColor: '#669AE0',
    width: 150,
    height: 45,
    borderColor: 'transparent',
    borderWidth: 0,
    borderRadius: 10,
  },
});

export default EntryDetail;
