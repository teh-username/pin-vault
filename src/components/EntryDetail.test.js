import React from 'react';
import EntryDetail, { ButtonRow } from './EntryDetail';

describe('EntryDetail Component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<EntryDetail name="name" code="code" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('ButtonRow Component', () => {
    it('renders as expected', () => {
      const wrapper = shallow(
        <ButtonRow onEditPress={() => {}} onDeletePress={() => {}} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
