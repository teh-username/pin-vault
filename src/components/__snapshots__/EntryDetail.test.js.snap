// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryDetail Component ButtonRow Component renders as expected 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "top": 30,
    }
  }
>
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "#669AE0",
        "borderColor": "transparent",
        "borderRadius": 10,
        "borderWidth": 0,
        "height": 45,
        "width": 150,
      }
    }
    icon={
      Object {
        "name": "mode-edit",
      }
    }
    onPress={[Function]}
    title="Edit Code"
  />
  <Button
    buttonStyle={
      Object {
        "backgroundColor": "#669AE0",
        "borderColor": "transparent",
        "borderRadius": 10,
        "borderWidth": 0,
        "height": 45,
        "width": 150,
      }
    }
    icon={
      Object {
        "name": "delete",
      }
    }
    onPress={[Function]}
    title="Delete Entry"
  />
</View>
`;

exports[`EntryDetail Component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "top": 30,
    }
  }
>
  <TextElement
    h1={true}
  >
    name
  </TextElement>
  <TextElement
    h1={true}
  >
    code
  </TextElement>
  <ButtonRow />
</View>
`;
